### PR TITLE
Dropped products prefetching from collections & categories resolvers

### PR DESCRIPTION
**Pull request by NyanKiyoshi**
This improves performances while fetching collections and categories.

### Pull Request Checklist

1. [ ] Privileged views and APIs are guarded by proper permission checks.
1. [ ] All visible strings are translated with proper context.
1. [ ] All data-formatting is locale-aware (dates, numbers, and so on).
1. [ ] Database queries are optimized and the number of queries is constant.
1. [ ] Database migration files are up to date.
1. [x] The changes are tested.
1. [ ] GraphQL schema and type definitions are up to date.
1. [x] Changes are mentioned in the changelog.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -111,10 +111,11 @@ All notable, unreleased changes to this project will be documented in this file.
 - Fix fetching staff user without `manage_users` permission - #4835 by @fowczarek
 - Ensure that a GraphQL query is a string - #4836 by @nix010
 - Add ability to configure the password reset link - #4863 by @fowczarek
+- Dropped products prefetch when resolving collections and categories - #5291 by @NyanKiyoshi
```
Comment (patrys) [inline CHANGELOG.md]: I don't think that's an informative changelog entry. As a user, I'd rather read something like "fixed a performance issue where Saleor would sometimes…"

**Comment by django-queries** [comment]
Here is the report for 8cb82111a4c5774263c8aec8467a62d8b3601e2d (NyanKiyoshi/saleor @ gql/prefetch/products)
Base comparison is 8a015911283dd06367d5ab1b5b6c5d699471e368.



<details><summary>No differences were found. (click me)</summary>
<p>

```diff
# api.benchmark checkout
  test name                                  	left count 	right count	duplicate count
  -------------------------------------------	-----------	-----------	---------------
  add billing address to checkout            	         34	         34	             20
  add shipping to checkout                   	          7	          7	              0
  checkout payment charge                    	         10	         10	              0
  complete checkout                          	          8	          8	              0
  create checkout                            	          5	          5	              1

# api.benchmark homepage
  test name                                  	left count 	right count	duplicate count
  -------------------------------------------	-----------	-----------	---------------
  retrieve main menu                         	          5	          5	              0
  retrieve product list                      	          4	          4	              0
  retrieve secondary menu                    	          5	          5	              0
  retrieve shop                              	          2	          2	              0

# api.benchmark product
  test name                                  	left count 	right count	duplicate count
  -------------------------------------------	-----------	-----------	---------------
  product details                            	         17	         17	              3
  retrieve product attributes                	          9	          9	              0

# api.benchmark variant
  test name                                  	left count 	right count	duplicate count
  -------------------------------------------	-----------	-----------	---------------
  product variant bulk create                	         51	         51	              3
  retrieve variant list                      	         23	         23	              9

# api product sorting attributes
  test name                                  	left count 	right count	duplicate count
  -------------------------------------------	-----------	-----------	---------------
  sort product not having attribute data     	         21	         21	              0
```

</p>
</details>

**Comment by patrys** [comment]
Should we add a test with measured queries?

**Comment by NyanKiyoshi** [comment]
I think we need to fix #4359 for it. The robot didn't notice any changes, where... it should have changed. Meaning there is not enough data inside the database to actual notice the changes.

**Pull request merged**